tests/DeepCopyTest/Reflection/ReflectionHelperTest.php

```diff
@@ -29,6 +29,40 @@ public function testMaintainPropertiesKey()
 
         $this->assertSame($expectedProps, array_keys($actualProps));
     }
+
+    /**
+     * @dataProvider propertyDataProvider
+     *
+     * @param string $name
+     */
+    public function testGetProperty($name)
+    {
+        $object = new ReflectionHelperTestChild();
+        $property = ReflectionHelper::getProperty($object, $name);
+        $this->assertInstanceOf('\ReflectionProperty', $property);
+        $this->assertSame($name, $property->getName());
+    }
+
+    /**
+     * @return array
+     */
+    public function propertyDataProvider()
+    {
+        return [
+            'public property' => ['a4'],
+            'private property' => ['a9'],
+            'private property of ancestor' => ['a3']
```

Comment:
@theofidry On one hand, accessing parent object clearly *is* violation of OOP encapsulation. But not supporting it during deep copy could result in violation of Liskov substitution principle, which is even worse (if your code is deigned to work with class `Y` and you subtitute it with class `Z` that inherits from `Y`, which is perfectly valid, it will suddenly break).

IMHO, it also contraditcs the intuition to use *DeepCopy* library in the first place, as it will not create completely isolated copy of the graph of objects, as shown on [this image](https://github.com/myclabs/DeepCopy/blob/master/doc/deep-copy.png).

My use-case was that I had graph of various objects linked together and I wanted to create a copy of this graph (while preserving inner references), try out some changes and then decise whether or not to "commit" these changes (replace the original graph of objects with the new one). Before I implemented #23, it was not creating the isolated copy of object graph and therefore it was useless for me.

```php
// this is how it worked before #23 was accepted
class X {}

class Y {
    private $x;
    public function __construct(X $x) { $this->x = $x; }
    public function getX() { return $this->x; }
}

class Z extends Y {}

$deepCopy = new DeepCopy;
$x = new X;

$y = new Y($x);
$yCopy = $deepCopy->copy($y);
var_dump($y->getX());
var_dump($yCopy->getX()); // X is copied as expected 

$z = new Z($x);
$zCopy = $deepCopy->copy($z);
var_dump($z->getX());
var_dump($zCopy->getX()); // X is not copied - violates Liskov substitution principle
```